src/main/java/com/github/technus/tectech/thing/metaTileEntity/multi/GT_MetaTileEntity_EM_infuser.java

```diff
@@ -164,13 +175,25 @@ public void outputAfterRecipe_EM() {
                 for (ItemStack itemStackInBus : inputBus.mInventory) {
                     if (itemStackInBus != null) {
                         Item item = itemStackInBus.getItem();
-                        if (itemStackInBus.stackSize == 1) {
-                            if (isItemStackFullyCharged(itemStackInBus)) {
+                        if (itemStackInBus.stackSize == 1 && item != null) {
+                            if (isItemStackFullyCharged(itemStackInBus) && isItemStackFullyRepaired(itemStackInBus)) {
                                 itemProcessed = true;
                                 if (addOutput(itemStackInBus)) {
                                     this.depleteInput(itemStackInBus);
                                 }
                             } else {
+                                if (item.isRepairable()) {
+                                    FluidStack uum = getStoredFluids().stream().filter(fluid -> Materials.UUMatter.getFluid(1).isFluidEqual(fluid)).findAny().orElse(null);
+                                    if (uum != null) {
+                                        int maxRepairedDamage = Math.max(item.getDamage(itemStackInBus), 1000);
+                                        int actualRepairedDamage = Math.min(item.getDamage(itemStackInBus) - maxRepairedDamage, 0);
+                                        if (getEUVar() >= actualRepairedDamage) {
```

Comment:
Redo logic here, these don't sound right.
Also 32767 EU is nothing, we should increase EU rate even though it consumes some UUM.